### ISSUE TITLE
pwfile keeps getting destroyed on `hab-sup` restart

### ISSUE THREAD

**Issue by bdangit** [habitat-sh/on-prem-builder#133]
Looks like on later versions of `hab-sup` (currently on `hab-sup/0.78.0/2019033123140`) , the `/hab/svc/*/config` directory is getting reconstructed, which means anything that was suppose to persist in there gets blown away.

If so, this is problematic with the current `habitat/builder-datastore` from `on-prem-stable` release as the `/hab/svc/builder-datastore/config/pwfile` keeps getting destroyed on my existing bldr setup.

I see from here https://github.com/habitat-sh/builder/blob/master/components/builder-datastore/hooks/init#L18-L22 that this logic has been added to always make sure the `pwfile` exists.

---

Currently running these versions:

```
habitat/builder-api/7987/20181203194923
habitat/builder-datastore/7809/20181019215440
habitat/builder-memcached/7728/20180929144821
habitat/builder-minio/7764/20181006010221
bdangit/builder-api-proxy/8031/20190111002925
```

I have a workaround which is to modify the currently installed version, but ideally we should have to just rely on the habitat approved versions.

**Comment by bdangit** [comment]
ugh turns out the line of code I referenced doesn't matter.  That is only to create the `pwfile` if the database does not exist.

I think the fix is to create the pwfile on init all the time.

**Comment by bdangit** [comment]
Actually - that file should exist in a more persistent location possible `var` or `data`

**Comment by chefsalim** [comment]
This will be resolved with release of hab 0.79 shortly
See PR - https://github.com/habitat-sh/habitat/pull/6348

**Comment by chefsalim** [comment]
Closing as 0.79.1 release is out there now - please reopen if you continue to see this behavior after hab update
